`--nbval-lax` cannot find notebook to test

While working on the deliverable report, I have just found this behaviour:

<pre>
(nbval) iota:examples fangohr$ py.test -v --nbval demo-with-fail.ipynb 
======================================= test session starts ========================================
platform darwin -- Python 3.6.0, pytest-3.0.6, py-1.4.32, pluggy-0.4.0 -- /Users/fangohr/anaconda/envs/nbval/bin/python
cachedir: .cache
rootdir: /Users/fangohr/git/OpenDreamKit-D48/WP4/D4.8/examples, inifile: 
plugins: nbval-0.4.1
collected 5 items 

demo-with-fail.ipynb::Cell 1 PASSED
demo-with-fail.ipynb::Cell 2 PASSED
demo-with-fail.ipynb::Cell 3 PASSED
demo-with-fail.ipynb::Cell 5 PASSED
demo-with-fail.ipynb::Cell 7 FAILED

============================================= FAILURES =============================================
______________________________________________ cell 7 ______________________________________________
Notebook cell execution failed
Cell 7: Cell outputs differ

Input:
datetime.datetime.now()

Traceback: mismatch 'text/plain'
<<<<<<<<<<<< Reference output from ipynb file:
datetime.datetime(2017, 2, 13, 10, 16, 15, 240125)
============ disagrees with newly computed (test) output:  
datetime.datetime(2017, 2, 13, 10, 43, 31, 39014)
>>>>>>>>>>>>
================================ 1 failed, 4 passed in 2.05 seconds ================================
</pre>
The above is okay and works as desired. Interesting is the following for a very minor change in calling the command: 

<pre>
(nbval) iota:examples fangohr$ py.test -v --nbval-lax demo-with-fail.ipynb 
======================================= test session starts ========================================
platform darwin -- Python 3.6.0, pytest-3.0.6, py-1.4.32, pluggy-0.4.0 -- /Users/fangohr/anaconda/envs/nbval/bin/python
cachedir: .cache
rootdir: /Users/fangohr/git/OpenDreamKit-D48/WP4/D4.8/examples, inifile: 
plugins: nbval-0.4.1
collecting 0 items
=================================== no tests ran in 0.00 seconds ===================================
ERROR: not found: /Users/fangohr/git/OpenDreamKit-D48/WP4/D4.8/examples/demo-with-fail.ipynb
(no name '/Users/fangohr/git/OpenDreamKit-D48/WP4/D4.8/examples/demo-with-fail.ipynb' in any of [])
</pre>

Looks like a bug?

--- a/nbval/plugin.py
+++ b/nbval/plugin.py
@@ -99,7 +99,8 @@ def pytest_collect_file(path, parent):
     """
     Collect IPython notebooks using the specified pytest hook
     """
-    if parent.config.option.nbval and path.fnmatch("*.ipynb"):
+    opt = parent.config.option
+    if (opt.nbval or opt.nbval_lax) and path.fnmatch("*.ipynb"):
         return IPyNbFile(path, parent)
 
 

--- a/nbval/plugin.py
+++ b/nbval/plugin.py
@@ -5,6 +5,8 @@ Authors: D. Cortes, O. Laslett, T. Kluyver, H. Fangohr, V.T. Fauske
 
 """
 
+from __future__ import print_function
+
 # import the pytest API
 import pytest
 import sys

--- a/tests/test_collect.py
+++ b/tests/test_collect.py
@@ -1,0 +1,71 @@
+
+from __future__ import print_function
+
+import os
+
+import nbformat
+
+
+pytest_plugins = "pytester"
+
+
+def _build_nb(sources):
+    """Builds a notebook of only code cells, from a list of sources"""
+    nb = nbformat.v4.new_notebook()
+    for src in sources:
+        nb.cells.append(nbformat.v4.new_code_cell(src))
+    return nb
+
+
+def _write_nb(sources, path):
+    nb = _build_nb(sources)
+    nbformat.write(nb, path)
+
+
+# Sources to try to collect
+sources = [
+    # In [1]:
+    "a = 5",
+    # In [2]:
+    "for i in range(10):\n" +
+    "    print(i)",
+    # In [3]:
+    "print(a)",
+    # In [4]:
+    "a",
+    # In [5]:
+    "import os\n" +
+    "os.curdir"
+]
+
+
+def test_collection_nbval(testdir):
+    # Write notebook to test dir
+    _write_nb(sources, os.path.join(str(testdir.tmpdir), 'test_collection.ipynb'))
+
+    # Run tests
+    items, recorder = testdir.inline_genitems('--nbval', '--current-env')
+
+    # Debug output:
+    for item in items:
+        print('Cell %d:' % item.cell_num)
+        print("    " + "\n    ".join(item.cell.source.splitlines()) + "\n")
+
+    # Checks:
+    assert len(items) == 5
+
+
+def test_collection_nbval_lax(testdir):
+    # Write notebook to test dir
+    _write_nb(sources, os.path.join(str(testdir.tmpdir), 'test_collection.ipynb'))
+
+    # Run tests
+    items, recorder = testdir.inline_genitems('--nbval-lax', '--current-env')
+
+    # Debug output:
+    for item in items:
+        print('Cell %d:' % item.cell_num)
+        print("    " + "\n    ".join(item.cell.source.splitlines()) + "\n")
+
+    # Checks:
+    assert len(items) == 5

